firewood/src/db.rs

```diff
@@ -390,7 +391,8 @@ impl api::Db for Db {
     type Proposal = Proposal;
 
     async fn revision(&self, root_hash: HashKey) -> Result<Arc<Self::Historical>, api::Error> {
-        if let Some(rev) = self.get_revision(&TrieHash(root_hash)) {
+        let rev = block_in_place(|| self.get_revision(&TrieHash(root_hash)));
+        if let Some(rev) = rev {
```

Comment:
```suggestion
        block_in_place(|| {
             self
                  .get_revision(&TrieHash(root_hash)))
                  .map(Arc::new)
                  .ok_or(api::Error::HashNotFound { provided: root_hash }
        });
```
That's my best guess at formatting

Reply:
nit ^